File: mario.c
#include <stdio.h>
#include <cs50.h>

int pyramid_height(void);

int main(void)
{
    const int n = pyramid_height();
    int width = n * 2 + 2;

    for (int y = 0; y < n; y++)
    {
        for (int x = 0; x < width; x++)
        {
            if (x == n || x == n + 1)
            {
                printf(" ");
            }
            else if (y + n + 2 < x)
            {
                printf(".");
            }
            else if (y + x < n - 1)
            {
                printf(".");
            }
            else
            {
                printf("#");
            }
        }
        printf("\n");
    }
}

int pyramid_height(void)
{
    int n;
    do
    {
        n = get_int("Pyramid height: ");
    }
    while (n < 1 || n > 8);
    return (n);
}

File: mario.py
from cs50 import get_int

while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break
    
width = height * 2 + 2

for y in range(height):
    for x in range(width):
        # Middle: print spaces 
        if x == height or x == height + 1:
            print(" ", end="")
        # Right side: height + 2 (brings us to right side) - 
        # If x is greater than y + height + 2 print nothing 
        elif y + height + 2 < x:
            print("", end="")
        # Left side: print spaces 
        elif y + x < height - 1:
            print(" ", end="")
        else:
            print("#", end="")
    print()